### PR TITLE
support Saved model

### LINKED ISSUE #17: Using Efficentdet with TF serving
Hi, has anyone tried to convert the efficentdet model to saved model format for tensorflow serving.

### LINKED ISSUE #23: Add saved model support
Hey @mingxingtan, I'm interested in the repo and I have implemented saved model feature which replaces ```_generate_detections``` by a TensorFlow graph code. Could I submit a PR?

### LINKED ISSUE #50: How to export a saved model ?
I found the output_node_names are a list, we should concat it to one tensor so that we can tranfer to saved model.　Have any methos?

['class_net/class-predict/BiasAdd', 'class_net/class-predict_1/BiasAdd', 'class_net/class-predict_2/BiasAdd', 
      'class_net/class-predict_3/BiasAdd', 'class_net/class-predict_4/BiasAdd', 'box_net/box-predict/BiasAdd', 'box_net/box-predict_1/BiasAdd',
       'box_net/box-predict_2/BiasAdd', 'box_net/box-predict_3/BiasAdd', 'box_net/box-predict_4/BiasAdd']

### PR DESCRIPTION
**Pull request by fsx950223**
Fix #50,Fix #23,Fix #17

### CODE REVIEW

**Comment by mingxingtan** [comment]
Thank you @fsx950223 , I saw you have pushed multiple changes. Are they ready to review now?

**Comment by fsx950223** [comment]
yes

File: efficientdet/anchors.py [hunk 1]
```diff
@@ -183,6 +216,75 @@ def _generate_anchor_boxes(image_size, anchor_scale, anchor_configs):
   anchor_boxes = np.vstack(boxes_all)
   return anchor_boxes
 
+def _generate_detections_tf(cls_outputs, box_outputs, anchor_boxes, indices,
+                         classes, image_id, image_scale, num_classes, use_native_nms=False):
```
Comment (mingxingtan) [inline efficientdet/anchors.py]: Format it to make them align. This line is also too long (please make it <80 characters).

File: efficientdet/anchors.py [hunk 2]
```diff
@@ -183,6 +216,75 @@ def _generate_anchor_boxes(image_size, anchor_scale, anchor_configs):
...
+
+  def _else(detections):
+    boxes_cls = tf.gather(boxes, indices)
+    scores_cls = tf.gather(scores, indices)
+    # Select top-scoring boxes in each class and apply non-maximum suppression
+    # (nms) for boxes in the same class. The selected boxes from each class are
+    # then concatenated for the final detection outputs.
+    all_detections_cls = tf.concat([tf.reshape(boxes_cls, [-1, 4]), scores_cls], axis=1)
+    if use_native_nms:
+      top_detection_idx = tf.image.non_max_suppression(all_detections_cls[:, :4],
```
Comment (mingxingtan) [inline efficientdet/anchors.py]: This is great! Did you verify both use_native_nms=True and False work?

Comment (fsx950223) [reply efficientdet/anchors.py]: Yes, both work well, but get different result.

**Comment by fsx950223** [comment]
> Thanks a lot for the PR. Overall looks great!
> 
> Have you verified this PR by running some commands?  If so, could you help add your command line? If everything works well, I will approve and merge this PR.

I add them later

**Comment by fsx950223** [comment]
> Thanks a lot for the PR. Overall looks great!
> 
> Have you verified this PR by running some commands? If so, could you help add your command line? If everything works well, I will approve and merge this PR.
Export saved model:
```python model_inspect.py --runmode=saved_model --model_name=efficientdet-d0 --ckpt_path=<ckpt_path> --input_image=<test_image_path> --output_image_dir=<output_image_path> --input_image_size=1920 --max_boxes_to_draw=100 --min_score_thresh=0.2```
Saved model inference:
```python model_inspect.py --runmode=saved_model_infer --model_name=efficientdet-d0 --ckpt_path=<ckpt_path> --input_image=<test_image_path> --output_image_dir=<output_image_path> --input_image_size=1920 --max_boxes_to_draw=100 --min_score_thresh=0.2```

**Pull request merged**